optuna/samplers/_grid.py

```diff
@@ -89,6 +89,11 @@ def objective(trial):
         suggested may vary. This is to reduce duplicate suggestions during distributed
         optimization.
 
+    Note:
+        When the trial is created by :meth:`~optuna.study.Study.enqueue_trial`,
```

Comment:
> We should specify all parameters with :meth:`~optuna.study.Study.enqueue_trial` in this case

It is a little bit unclear to me for `this case` here. I'm not confident but I feel this note has two topics: (a) trial created by enqueue_trial is independent on `GridSampler`, and (b) it requires all parameters specified if users want to use enqueue_trial with `GridSampler`. If it is correct, how about dividing this note into two ones?

Reply:
Your suggestion seems reasonable, but there are too many Note sections already. How about keeping the single one and replacing a more precise explanation? For example, "We should specify all parameters with `enqueue_trial` when using `GridSampler`".

Reply:
> but there are too many Note sections already
> How about keeping the single one and replacing a more precise explanation?

I agree with you. The suggestion makes documentation more natural to me.